zigpy/types/basic.py

```diff
@@ -236,11 +251,20 @@ def serialize(self):
 
     @classmethod
     def deserialize(cls, data):
+        if len(data) < cls._prefix_length:
+            raise ValueError('Data is too short')
+
         length = int.from_bytes(data[:cls._prefix_length], 'little')
-        raw = data[cls._prefix_length:length + 1]
-        r = cls(raw.split(b'\x00')[0].decode('utf8', errors='replace'))
+
+        if len(data) < cls._prefix_length + length:
+            raise ValueError('Data is too short')
+
+        raw = data[cls._prefix_length:cls._prefix_length + length]
+        raw = raw.split(b'\x00')[0]
```

Comment:
Don't split raw by null byte. Need the entire thing as raw, however input into `decode('utf8')` should be split

Reply:
Only because xiaomi are sending bytes in `CharacterString` ZCL type.